core/src/main/java/com/amplitude/core/platform/intercept/IdentifyInterceptFileStorageHandler.kt

```diff
@@ -0,0 +1,118 @@
+                if (eventsString.isEmpty()) continue
+                val eventsList = JSONArray(eventsString).toEvents()
+                var events = eventsList
+                if (event == null) {
+                    event = eventsList[0]
+                    identifyEventUserProperties = event.userProperties!![IdentifyOperation.SET.operationType] as MutableMap<String, Any?>
+                    events = eventsList.subList(0, eventsList.size)
+                }
+                val userProperties = IdentifyInterceptorUtil.mergeIdentifyList(events)
+                identifyEventUserProperties?.putAll(userProperties)
+                removeFile(eventPath as String)
+            } catch (e: JSONException) {
+                logger.warn("Identify Merge error: " + e.message)
+                removeFile(eventPath as String)
+            }
+            event?.userProperties!![IdentifyOperation.SET.operationType] =
+                identifyEventUserProperties
+        }
+        return event
+    }
+
+    override suspend fun fetchAndMergeToNormalEvent(event: BaseEvent): BaseEvent {
+        val eventsData = storage.readEventsContent()
+        if (eventsData.isEmpty()) {
+            return event
+        }
+        val userProperties = mutableMapOf<String, Any?>()
+        for (eventPath in eventsData) {
+            try {
+                val eventsString = storage.getEventsString(eventPath)
+                if (eventsString.isEmpty()) continue
+                val events = JSONArray(eventsString).toEvents()
+                val listUserProperties = IdentifyInterceptorUtil.mergeIdentifyList(events)
+                userProperties.putAll(listUserProperties)
+                removeFile(eventPath as String)
+            } catch (e: JSONException) {
+                logger.warn("Identify Merge error: " + e.message)
+                removeFile(eventPath as String)
+            }
+            event.userProperties?.let {
+                userProperties.putAll(it)
+            }
+            event.userProperties = userProperties
+        }
+        return event
+    }
+
+    override suspend fun fetchAndMergeToIdentifyEvent(event: BaseEvent): BaseEvent {
+        val eventsData = storage.readEventsContent()
+        if (eventsData.isEmpty()) {
+            return event
+        }
+        val userProperties = mutableMapOf<String, Any?>()
+        for (eventPath in eventsData) {
+            try {
+                val eventsString = storage.getEventsString(eventPath)
+                if (eventsString.isEmpty()) continue
+                val events = JSONArray(eventsString).toEvents()
+                val listUserProperties = IdentifyInterceptorUtil.mergeIdentifyList(events)
+                userProperties.putAll(listUserProperties)
+                removeFile(eventPath as String)
+            } catch (e: JSONException) {
+                logger.warn("Identify Merge error: " + e.message)
+                removeFile(eventPath as String)
+            }
```

Comment:
nit: There is a decent amount of duplicated code between `fetchMergeToIdentifyEvent` and `fetchMergeToNormalEvent`. Could be nice to centralize loading/merging in a single place.

Reply:
Good suggestion, created a helper function for it.